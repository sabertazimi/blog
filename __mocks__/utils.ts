let navigatorClipboardMocked = false
const writeTextMock = jest.fn()

const mockNavigatorClipboard = () => {
  if (!navigatorClipboardMocked) {
    Object.defineProperty(global.navigator, 'clipboard', {
      value: {
        writeText: writeTextMock,
      },
    })

    navigatorClipboardMocked = true
  }

  return { writeTextMock }
}

export { mockNavigatorClipboard }
