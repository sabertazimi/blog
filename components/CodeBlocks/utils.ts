import parseNumericRange from 'parse-numeric-range'
import type { Language } from 'prism-react-renderer'

const normalizeCode = (code: string = '') => code.replace(/\n+$/, '')

function isLanguage(language: string): language is Language {
  return language.length > 0
}

function normalizeLanguage(languageClass?: string): Language {
  if (languageClass != null) {
    const language = languageClass.replace('language-', '')

    if (isLanguage(language))
      return language
  }

  return 'typescript'
}

function normalizeLanguageName(language: string): string {
  switch (language) {
    case 'html':
      return 'HTML'
    case 'xml':
      return 'XML'
    case 'yml':
    case 'yaml':
      return 'YAML'
    case 'css':
      return 'CSS'
    case 'json':
      return 'JSON'
    case 'md':
    case 'markdown':
      return 'Markdown'
    case 'js':
    case 'javascript':
      return 'JavaScript'
    case 'ts':
    case 'typescript':
      return 'TypeScript'
    case 'coffee':
    case 'coffeescript':
      return 'CoffeeScript'
    case 'jsx':
    case 'tsx':
      return 'React'
    case 'objc':
    case 'objectivec':
      return 'Objective-C'
    default:
      return language.charAt(0).toUpperCase() + language.slice(1)
  }
}

function normalizeLines(expression: string): Set<number> {
  return new Set(parseNumericRange(expression))
}

export {
  normalizeCode,
  normalizeLanguage,
  normalizeLanguageName,
  normalizeLines,
}
