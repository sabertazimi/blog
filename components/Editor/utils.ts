function languageToFilepath(language?: string): string {
  switch (language) {
    case 'css':
      return '/styles.css'
    case 'js':
    case 'javascript':
      return '/App.js'
    case 'ts':
    case 'typescript':
      return '/App.ts'
    case 'jsx':
      return '/App.jsx'
    case 'tsx':
      return '/App.tsx'
    case 'vue':
      return '/src/App.vue'
    case undefined:
    default:
      return '/App.tsx'
  }
}

function normalizeFilepath(filename?: string, language?: string): string {
  if (filename == null || filename === '')
    return languageToFilepath(language)

  if (!filename.startsWith('/'))
    return `/${filename}`

  return filename
}

export { normalizeFilepath }
