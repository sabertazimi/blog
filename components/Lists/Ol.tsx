import cx from 'classnames'
import type { DetailedHTMLProps, OlHTMLAttributes } from 'react'
import styles from './Ol.module.css'

interface Props
  extends DetailedHTMLProps<
    OlHTMLAttributes<HTMLOListElement>,
    HTMLOListElement
  > {}

function Ol({ children, className, ...props }: Props): JSX.Element {
  return (
    <ol {...props} className={cx(className, styles.ol)}>
      {children}
    </ol>
  )
}

export default Ol
