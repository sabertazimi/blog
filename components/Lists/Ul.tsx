import type { DetailedHTMLProps, OlHTMLAttributes } from 'react'
import cx from 'classnames'
import styles from './Ul.module.css'

interface Props
  extends DetailedHTMLProps<
    OlHTMLAttributes<HTMLUListElement>,
    HTMLUListElement
  > {}

function Ul({ children, className, ...props }: Props): JSX.Element {
  return (
    <ul {...props} className={cx(className, styles.ul)}>
      {children}
    </ul>
  )
}

export default Ul
