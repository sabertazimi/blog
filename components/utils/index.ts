import classNames from 'classnames';

export { classNames };
