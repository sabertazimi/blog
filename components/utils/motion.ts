export { AnimatePresence, motion } from 'framer-motion';
export type { MotionProps } from 'framer-motion';
