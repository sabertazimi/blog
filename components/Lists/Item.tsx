import { ArrowRight } from '@components/Icons'
import { cx } from '@components/utils'
import { siteConfig } from '@config'
import type { HTMLProps } from 'react'
import styles from './Item.module.css'

interface Props extends HTMLProps<HTMLLIElement> {}

function Item({ children, className, ...props }: Props): JSX.Element {
  return (
    <li {...props} className={cx(className, styles.item, 'dark:text-light')}>
      <ArrowRight style={{ color: siteConfig.themeColor }} />
      <div>{children}</div>
    </li>
  )
}

export default Item
