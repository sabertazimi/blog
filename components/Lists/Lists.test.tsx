import { render } from '@/utils'
import Item from './Item'
import Ol from './Ol'
import Ul from './Ul'

describe('Lists', () => {
  const Lists = {
    ul: Ul,
    ol: Ol,
    li: Item,
  }

  it.each(Object.values(Lists))(
    'should render correctly (snapshot)',
    (List) => {
      const { container } = render(<List>List</List>)

      expect(container).toMatchSnapshot()
    },
  )
})
