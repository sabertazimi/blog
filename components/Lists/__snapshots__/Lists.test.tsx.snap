// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lists should render correctly (snapshot) 1`] = `
<div>
  <ul
    class="ul"
  >
    List
  </ul>
</div>
`;

exports[`Lists should render correctly (snapshot) 2`] = `
<div>
  <ol
    class="ol"
  >
    List
  </ol>
</div>
`;

exports[`Lists should render correctly (snapshot) 3`] = `
<div>
  <li
    class="item dark:text-light"
  >
    <span
      aria-label="arrow-right"
      class="anticon anticon-arrow-right"
      role="img"
      style="color: rgb(22, 119, 255);"
    >
      <svg
        aria-hidden="true"
        data-icon="arrow-right"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M869 487.8L491.2 159.9c-2.9-2.5-6.6-3.9-10.5-3.9h-88.5c-7.4 0-10.8 9.2-5.2 14l350.2 304H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h585.1L386.9 854c-5.6 4.9-2.2 14 5.2 14h91.5c1.9 0 3.8-.7 5.2-2L869 536.2a32.07 32.07 0 000-48.4z"
        />
      </svg>
    </span>
    <div>
      List
    </div>
  </li>
</div>
`;
